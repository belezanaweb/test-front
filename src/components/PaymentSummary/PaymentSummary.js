import React, { useContext } from 'react'
import { createUseStyles } from 'react-jss'
import { FormContext } from '../../components/Context/FormContext'
import HeaderText from '../HeaderText/HeaderText'
import Card from '../Card/Card'

const useStyles = createUseStyles({
  paymentSummary: {
    textTransform: 'uppercase',
    width: '100%',
    minWidth: '320px',
    maxWidth: '600px',
    margin: '0 auto'
  },
  summaryRow: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: '26px',
    textTransform: 'uppercase'
  },
  discount: {
    color: '#FF6C00'
  },
  total: {
    marginTop: '10px',
    fontWeight: 'bold'
  }
})

const PaymentSummary = () => {
  const classes = useStyles()
  const formContext = useContext(FormContext)

  const renderSummaryRows = () => {
    return (
      <>
        <div className={classes.summaryRow}>
          <span>{`****.****.****.${formContext.formFieldsValues.creditCard.slice(12)}`}</span>
        </div>
        <div className={classes.summaryRow}>
          <span>{formContext.formFieldsValues.holderName}</span>
        </div>
        <div className={classes.summaryRow}>
          <span>{formContext.formFieldsValues.expiryDate}</span>
        </div>
      </>
    )
  }

  return (
    <>
      <HeaderText>Pagamento</HeaderText>
      <div className={classes.paymentSummary}>
        <Card>{renderSummaryRows()}</Card>
      </div>
    </>
  )
}

export default PaymentSummary
