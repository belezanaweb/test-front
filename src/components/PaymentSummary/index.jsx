import React from 'react'

import { Container, Text } from './styles'

function PaymentSummary() {
  return (
    <Container>
      <Text>****.****.****.1234</Text>
      <Text>Jos√© da Silva</Text>
      <Text>05/2019</Text>
    </Container>
  )
}

export default PaymentSummary
