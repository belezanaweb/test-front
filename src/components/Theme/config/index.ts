export { default as light } from "./light";
