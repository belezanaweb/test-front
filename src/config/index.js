export const cartSteps = ['Sacola', 'Pagamento', 'Confirmação']
