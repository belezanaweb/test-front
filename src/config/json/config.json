{
    "ENDPOINT_BASKET": "https://www.mocky.io/v2/5b15c4923100004a006f3c07"
}