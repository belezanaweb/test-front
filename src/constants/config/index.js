export const BASE_URL = 'http://www.mocky.io/v2/'
export const BAG_ID = '5b15c4923100004a006f3c07'
