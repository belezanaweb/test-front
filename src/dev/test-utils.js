// source: https://testing-library.com/docs/react-testing-library/setup#custom-render
import React, { Suspense } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'emotion-theming'
import theme from '../theme'

// eslint-disable-next-line react/prop-types
function WithProviders ({ children }) {
  return (
    <ThemeProvider theme={theme}>
      <Suspense fallback={<>Loading...</>}>{children}</Suspense>
    </ThemeProvider>
  )
}

const customRender = (ui, options) => render(ui, { wrapper: WithProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
