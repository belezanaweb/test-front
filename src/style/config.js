// CONVERT TO REM
export const rem = (number) => {
  let tmp = number
  tmp = typeof number === 'string' && number.indexOf('px') ? number.replace('px', '') : number
  tmp = parseFloat(tmp)

  return `${tmp / 16}rem` // Minimum to font
}