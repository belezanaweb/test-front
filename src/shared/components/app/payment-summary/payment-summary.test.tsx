import React from 'react';
import { render, screen } from '../../../../../config/test-utils';

import PaymentSummary from './index';

const mock = {
  creditCardNumber: '****.****.****.1234',
  name: 'Jos√© da Silva',
  expires: '05/2019'
};

describe('<PaymentSummary />', () => {
  it('should render component with <div> tag', () => {
    render(<PaymentSummary {...mock} />, {});
    expect(screen.findByRole('div'));
  })

  it('should render component with default props and match snapshot', () => {
    const { container } = render(<PaymentSummary {...mock} />, {});
    expect(container.firstChild).toMatchSnapshot();
  })
})
