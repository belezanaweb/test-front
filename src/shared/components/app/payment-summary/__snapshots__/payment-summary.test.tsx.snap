// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentSummary /> should render component with default props and match snapshot 1`] = `
.c0 {
  color: #000;
  font-size: 1.4rem;
  font-weight: 400;
  line-height: 1.2;
  outline: none;
  text-transform: uppercase;
}

.c1 {
  color: #ff7800;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 1.4rem;
  font-weight: 700;
  margin-bottom: 2rem;
  line-height: 1.2;
  outline: none;
  text-transform: uppercase;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 2px solid #ff7800;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  fill: #ff7800;
  height: 4rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 1rem;
  width: 4rem;
}

.c4 {
  background: #fff;
  border-radius: 0.3rem;
  box-shadow: 1px 1px 5px 0 #00001d38;
  padding: 1rem;
}

.c3 {
  color: #999;
  font-size: 1.4rem;
  font-weight: 700;
  margin-bottom: 0.5rem;
  text-transform: uppercase;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <svg
        fill="#ff7800"
        viewBox="0 0 100 70.05"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M98.59 1.41a4.816 4.816 0 00-6.81 0L38.65 62.07 8.21 36.53a4.816 4.816 0 00-6.81 6.81l32.9 25.55c1.88 1.88 7.37 1.31 9.25-.56L98.59 8.21c1.88-1.87 1.88-4.92 0-6.8z"
        />
      </svg>
    </div>
    Compra Efetuada com sucesso
  </div>
  <div>
    <h3
      class="c3"
      role="heading"
    >
      Pagamento
    </h3>
    <div
      class="c4"
    >
      <ul>
        <li>
          ****.****.****.1234
        </li>
        <li>
          Jos√© da Silva
        </li>
        <li>
          05/2019
        </li>
      </ul>
    </div>
  </div>
</div>
`;
